src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -418,9 +436,17 @@ public void Reverse()
         public void Reverse(int index, int count)
         {
             if (index < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(index));
+            }
             if (count < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            if (Count - index < count)
+            {
+                throw new ArgumentException(SR.ArgumentInvalidOffLen);
+            }
```

Comment:
I see.  So this is actually a functional bug fix, where if we didn't add this check, we could end up corrupting the builder?  Makes sense.